### ISSUE TITLE
Orphan processes allow multiple autosubmit run expid

### ISSUE THREAD

**Issue by kinow** [BSC-ES/autosubmit#1304]
In GitLab by @franra9 on May 23, 2024, 16:50

( This is a template for bug reports. Please fill in the relevant information and delete the rest. )

Hello @dbeltrankyl and @kinow,

### Autosubmit Version

4.1.6
### Expid affected(If applicable)

a1be

### Which task has issues? Where is the log(If applicable)

* **Full_name**: autosubmit run expid

* **Log_Path:** -

### Summary

after seeing the processes in the lumi VM, I notice that there are old processes there from yesterday, but they don't impede me to launch again autosubmit run.

### Steps to reproduce

(How one can reproduce the issue - this is very important)

### What is the current bug behavior?

Not sure. I run the experiments in nohup most of the times.

### What is the expected correct behavior?

AS not allowing to run another AS run expid with the same expid, or to kill the zombie processes in the VM after some run in cancelled (sometimes yesterday I executed AS stop)

### Relevant logs and/or screenshots(if applicable)

```
[froura@climatedt-wf a1be]$ ps aux | grep a1be
froura   3138371  0.0  0.1 618412 31332 ?        Sl   May22   0:46 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3160959  0.0  0.4 553132 71752 ?        Sl   May22   0:43 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3160978  0.0  0.4 618412 70336 ?        Sl   May22   0:42 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3160979  0.0  0.4 618668 68368 ?        S    May22   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3296295  0.0  0.5 552828 84688 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3296300  0.0  0.5 618364 84540 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3297170  0.0  0.5 552572 84428 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3297177  0.0  0.5 618620 86168 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3297185  0.0  0.5 618364 82968 ?        S    May22   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3302262  0.0  0.5 553084 86112 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3302277  0.0  0.5 552828 82944 ?        S    May22   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3302280  0.0  0.5 618108 84568 ?        Sl   May22   0:35 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3628497  0.0  0.5 552588 82736 pts/34   Sl   11:52   0:09 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3628498  0.0  0.5 552844 80852 pts/34   S    11:52   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3628503  0.0  0.5 618636 87088 pts/34   Sl   11:53   0:09 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839379  0.0  0.5 552844 86384 pts/34   Sl   15:52   0:02 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839388  0.0  0.5 618636 90308 pts/34   Sl   15:52   0:02 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839724  1.2  0.6 694416 104756 pts/34  Sl   15:53   1:26 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839747  0.0  0.5 552588 86536 pts/34   Sl   15:53   0:02 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839748  0.0  0.5 552844 84040 pts/34   S    15:53   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3839753  0.0  0.5 618636 88900 pts/34   Sl   15:53   0:02 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3867431  1.4  0.6 915884 109324 pts/34  Sl   17:16   0:26 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be
froura   3867463  0.0  0.5 552860 87488 pts/34   Sl   17:16   0:00 /appl/AS/4.1.6/bin/python /appl/AS/4.1.6/bin/autosubmit run a1be

```

### Any other relevant information(if applicable)

()

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 24, 2024, 10:09

Hello @franra9 ,

How did you stop the experiment? This shouldn't happen as now the processors are a daemon

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on May 24, 2024, 10:22

run one
```
(debug_pip) (base) dbeltran@ bsces107894:~/new_autosubmit/a053/tmp$ ps ax | grep -i "a053"
  12558 pts/3    Ssl+   0:04 /home/dbeltran/debug_pip/bin/python /home/dbeltran/PycharmProjects/autosubmit_2022a/bin/autosubmit -lc DEBUG -lf DEBUG run a053
  12570 pts/3    Sl+    0:00 /home/dbeltran/debug_pip/bin/python /home/dbeltran/PycharmProjects/autosubmit_2022a/bin/autosubmit -lc DEBUG -lf DEBUG run a053
```

Used pycharm stop ( control + c equivalent):
```
(debug_pip) (base) dbeltran@ bsces107894:~/new_autosubmit/a053/tmp$ ps ax | grep -i "a053"
  12587 pts/1    S+     0:00 grep --color=auto -i a053
```

New run 
```
(debug_pip) (base) dbeltran@ bsces107894:~/new_autosubmit/a053/tmp$ ps ax | grep -i "a053"
  12598 pts/3    Ssl+   0:03 /home/dbeltran/debug_pip/bin/python /home/dbeltran/PycharmProjects/autosubmit_2022a/bin/autosubmit -lc DEBUG -lf DEBUG run a053
  12610 pts/3    Sl+    0:00 /home/dbeltran/debug_pip/bin/python /home/dbeltran/PycharmProjects/autosubmit_2022a/bin/autosubmit -lc DEBUG -lf DEBUG run a053
  12615 pts/1    S+     0:00 grep --color=auto -i a053
```
Used kill
```
(debug_pip) (base) dbeltran@ bsces107894:~/new_autosubmit/a053/tmp$ kill 12598
(debug_pip) (base) dbeltran@ bsces107894:~/new_autosubmit/a053/tmp$ ps ax | grep -i "a053"
  12617 pts/1    S+     0:00 grep --color=auto -i a053
```

Not sure how to reproduce the zombie processor

**Comment by kinow** [comment]
In GitLab by @franra9 on May 24, 2024, 10:55

If it happens again I'll post it here. Unfortunately I can not remember 100% how did I stop the autosubmit instance.

**Comment by kinow** [comment]
This might have been fixed with 4.1.9. There is another issue with portalocker where we can have multiple run expid... and another one with zombies (fixed).... since the last comment is from 2 months, worth checking with latest.

**Comment by kinow** [comment]
In GitLab by @dbeltrankyl on Aug 5, 2024, 11:58

@kinow , I think we can close this one. I'm sure this is fixed in v4.1.9 ( the orphan ones), and for Portalocker, there is an issue for #945. There is no relation between zombies/orphans and multiple runs

- 4.1.9 fixed orphan processors # will rename the title
- 4.1.9 introduced zombie processors #1381
- 4.1.10 will fix multiple runs ( hopefully)

So we can continue in #945

**Comment by kinow** [comment]
Thank you, Dani!
